discovery-infra/start_discovery.py

```diff
@@ -409,28 +413,51 @@ def nodes_flow(client, cluster_name, cluster, machine_net, kube_client=None, clu
     main_cidr = args.vm_network_cidr if is_ipv4 else args.vm_network_cidr6
     secondary_cidr = machine_net.provisioning_cidr_v4 if is_ipv4 else machine_net.provisioning_cidr_v6
 
+    nodes_number = args.master_count + args.number_of_workers
+
     if client:
         cluster_info = client.cluster_get(cluster.id)
         macs = utils.get_libvirt_nodes_macs(nodes_details["libvirt_network_name"])
         if is_none_platform_mode():
             macs += utils.get_libvirt_nodes_macs(nodes_details["libvirt_secondary_network_name"])
 
         if not (cluster_info.api_vip and cluster_info.ingress_vip):
-            utils.wait_till_hosts_with_macs_are_in_status(
-                client=client,
-                cluster_id=cluster.id,
-                macs=macs,
-                statuses=[
-                    consts.NodesStatus.INSUFFICIENT,
-                    consts.NodesStatus.PENDING_FOR_INPUT,
-                    consts.NodesStatus.KNOWN
-                ],
-            )
+            if args.kube_api:
+                cluster_deployment.wait_for_agents(nodes_number)
+            else:
+                utils.wait_till_hosts_with_macs_are_in_status(
+                    client=client,
+                    cluster_id=cluster.id,
+                    macs=macs,
+                    statuses=[
+                        consts.NodesStatus.INSUFFICIENT,
+                        consts.NodesStatus.PENDING_FOR_INPUT,
+                        consts.NodesStatus.KNOWN
+                    ],
+                )
 
             if args.master_count == 1:
                 tf.change_variables({"single_node_ip": helper_cluster.Cluster.get_ip_for_single_node(
                     client, cluster.id, main_cidr, ipv4_first=is_ipv4)})
-                set_cluster_machine_cidr(client, cluster.id, machine_net, set_vip_dhcp_allocation=False)
+                if args.kube_api:
+                    cluster_deployment.patch(
+                        install_strategy=InstallStrategy(
+                            host_prefix=args.host_prefix if is_ipv4 else args.host_prefix6,
+                            machine_cidr=get_machine_cidr_from_machine_net(machine_net),
+                            cluster_cidr=args.cluster_network if is_ipv4 else args.cluster_network6,
+                            service_cidr=args.service_network if is_ipv4 else args.service_network6,
+                            ssh_public_key=args.ssh_key,
+                            control_plane_agents=args.master_count,
+                            worker_agents=args.number_of_workers,
+                        ),
+                    )
```

Comment:
Do we even need this patch?
I thought we already set all those fields with the right value
Alternatively, we must re-specify all the fields here?

Reply:
you're right, this is unnecessary